pkg/bootkube/parse.go

```diff
@@ -0,0 +1,67 @@
+package bootkube
+
+import (
+	"io/ioutil"
+	"path/filepath"
+	"strings"
+
+	"github.com/ghodss/yaml"
+	"github.com/kubernetes-incubator/bootkube/pkg/asset"
+	"k8s.io/kubernetes/pkg/apis/extensions/v1beta1"
+)
+
+const (
+	defaultServiceCIDR = "10.3.0.0/24"
+	defaultPodCIDR     = "10.2.0.0/24"
+)
+
+// detectServiceCIDR deserializes the '--service-cluster-ip-range' from the
+// kube-apiserver manifest
+func detectServiceCIDR(config Config) string {
+	b, err := ioutil.ReadFile(filepath.Join(config.AssetDir, asset.AssetPathAPIServer))
+	if err != nil {
+		UserOutput("Can't detect --service-cluster-ip-range in %s, using %s: %v\n", asset.AssetPathAPIServer, defaultServiceCIDR, err)
+		return defaultServiceCIDR
+	}
+	var apiServer v1beta1.DaemonSet
+	yaml.Unmarshal(b, &apiServer)
```

Comment:
We shouldn't ignore error here. Maybe the manifest is a deployment rather than daemonset (and unmarshal fails). Similar to above we should surface this information to the user.